external-crates/move/move-bytecode-verifier/src/type_safety.rs

```diff
@@ -863,20 +881,18 @@ fn verify_instr(
         }
 
         Bytecode::VecUnpack(idx, num) => {
-            let operand_vec = safe_unwrap!(verifier.stack.pop());
+            let operand_vec = safe_unwrap_err!(verifier.stack.pop());
             let declared_element_type = &verifier.resolver.signature_at(*idx).0[0];
             if operand_vec != ST::Vector(Box::new(declared_element_type.clone())) {
                 return Err(verifier.error(StatusCode::TYPE_MISMATCH, offset));
             }
-            for _ in 0..*num {
-                verifier.push(meter, declared_element_type.clone())?;
-            }
+            verifier.push_n(meter, declared_element_type.clone(), *num)?;
```

Comment:
This changes metering behavior from `ty_push_cost * n` to just `ty_push_cost`. re we ok with this change? I'd be more comfortable with charging ` * n` if possible...

Reply:
As in keeping the behavior from before the PR